File: 矩阵相乘/matrix.py
# encoding: utf-8

from random import  *
from  time  import  *

bef = 0

def gmatrix(m, n):
  r = []
  for i in range(m):
    row = []
    for j in range(n):
      row.append(random())
    r.append(row)
  return r

left = gmatrix(100, 500)
right = gmatrix(500, 100)

def multiply():
  global left
  global right
  m = len(left)
  n = len(right)
  l = len(right[0])

  r = []
  for i in range(m):
    row = []
    for j in range(l):
      t = 0
      for k in range(n):
        t += (left[i][k] * right[k][j])
      row.append(t)
    r.append(row)
  return r

def loop1000():
  i = 0
  global bef
  bef = int(time()*1000)
  while i < 1000:
    r = multiply()
    i += 1

loop1000()

print(str(int(time()*1000) - bef))

File: 矩阵相乘/matrix.js
function gmatrix(m, n) {
  var r = [];
  for(var i = 0; i < m; ++i) {
  	const row = [];
  	for(var j = 0; j < n; ++j) {
  	  row.push(Math.random());
    }
    r.push(row);
  }
  return r;
}

const a = gmatrix(100, 500);
const b = gmatrix(500, 100);

function multiply(left, right) {
   const m = left.length;
   const n = right.length;
   const l = right[0].length;

   const r = [];
   for(var i = 0; i < m; ++i) {
     const row = [];
     for(var j = 0; j < l; ++j) {
       let t = 0;
       for(var k = 0; k < n; ++k) {
         t += left[i][k] * right[k][j];
       }
       row.push(t);
     }
     r.push(row);
   }
   return r;
}

bec = new Date().getTime();

for (var i = 0; i < 1000; ++i) {
  multiply(a, b)
}

console.log('花费时间(毫秒) ' + ((new Date()).getTime() - bec));
